components/Tools/Generators/RandomString/index.tsx

```diff
@@ -0,0 +1,184 @@
+        .map(() => {
+          return range(1, optLength)
+            .map(() => {
+              // pick random char from pool
+              return pool.charAt(Math.floor(Math.random() * len));
+            })
+            .join("");
+        })
+        .join("\n");
+      setOutput(out);
+    };
+    f();
+  }, [optLength, optRow, flags]);
+
+  return (
+    <VStack spacing="5">
+      {/* Output Box */}
+      <Box flexGrow="1" w="100%">
+        <Flex>
+          <Text fontSize="lg" ml="45%" marginLeft={"10px"}>
+            Output
+          </Text>
+          <HStack spacing="2" marginLeft={"auto"}>
+            <CopyIconButton
+              ariaLabel="Copy"
+              icon={<FiCopy />}
+              onCopy={onCopy}
+              hasCopied={hasCopied}
+            />
+          </HStack>
+        </Flex>
+        <Controlled
+          onBeforeChange={() => {}}
+          value={output}
+          options={{
+            lineNumbers: true,
+            readOnly: true,
+            mode: "text",
+            theme: "material",
+          }}
+        />
+      </Box>
+      {/* Configuration part */}
+      <Flex w="100%">
+        <VStack spacing="3">
+          <FormControl as="fieldset">
+            <FormLabel as="legend">Length</FormLabel>
+            <NumberInput
+              value={optLength}
+              min={1}
+              max={maxLength}
+              onChange={handleOptLengthChange}
+            >
+              <NumberInputField />
+              <NumberInputStepper>
+                <NumberIncrementStepper />
+                <NumberDecrementStepper />
+              </NumberInputStepper>
+            </NumberInput>
+            <FormLabel as="legend">Row</FormLabel>
+            <NumberInput
+              value={optRow}
+              min={1}
+              max={maxLength}
+              onChange={handleOptRowChange}
+            >
+              <NumberInputField />
+              <NumberInputStepper>
+                <NumberIncrementStepper />
+                <NumberDecrementStepper />
+              </NumberInputStepper>
+            </NumberInput>
+            {/* charactor settings */}
+            <FormLabel as="legend">Charactors</FormLabel>
+            <VStack spacing={2} align={"left"}>
+              <CheckboxGroup
+                defaultValue={flags}
+                onChange={value => setFlags(value as string[])}
+              >
+                <Checkbox defaultIsChecked value="number">
+                  0-9
+                </Checkbox>
+                <Checkbox defaultIsChecked value="lower">
+                  a-z
+                </Checkbox>
+                <Checkbox defaultIsChecked value="upper">
+                  A-Z
+                </Checkbox>
+                <Checkbox defaultIsChecked value="special">
+                  !@#$%^&*
+                </Checkbox>
+                <Checkbox value="avoid">
+                  Avoid misreading chars (iloqILOQ019!)
```

Comment:
I didn't understand what this does, can you tell me what it is and make it clearer in the frontend?

Reply:
fixed!

Reply:
This is a setting to avoid using strings of characters that can be easily misread, and is useful when setting passwords. What kind of wording do you think is easy to understand?  "Avoid strings easy to mistake (ex: iloqILOQ019!)" something like this?

Reply:
Something like "Avoid letters that can be mistaken (e.g.: iloqILOQ019!) should be good. Then I will merge the pr

Reply:
ok, fixed!